### PR TITLE
feat: change condition of duplicate check

### LINKED ISSUE #4268: Change duplicate formula from ILIKE to strict match
https://github.com/twentyhq/twenty/blob/main/packages/twenty-server/src/workspace/workspace-query-builder/factories/find-duplicates-query.factory.ts#L123

I think we should change this to a strict equality check because it causes false positive.

Also I've received an email from one our users who had many false matches, their contacts that had "-" as LinkedIn URLs instead of leaving the URL empty. I feel like we could also add a min of 3 letters for it to be taken into account, but I'm not 100% sure it's a good idea. What do you think @rostaklein? Do you see any downside to it?

### PR DESCRIPTION
**Pull request by rostaklein**
fixes https://github.com/twentyhq/twenty/issues/4268

### CODE REVIEW

File: packages/twenty-server/src/workspace/workspace-query-builder/factories/find-duplicates-query.factory.ts [hunk 1]
```diff
@@ -113,14 +113,20 @@ export class FindDuplicatesQueryFactory {
...
+        return argExists && isArgLengthValid;
+      }),
     );
 
     const filterCriteria = criteriaWithMatchingArgs.map((criteria) =>
       Object.fromEntries(
         criteria.columnNames.map((columnName) => [
           columnName,
-          { ilike: `%${argsData[columnName]}%` },
+          { eq: `%${argsData[columnName]}%` },
```
Comment (FelixMalfait) [inline packages/twenty-server/src/workspace/workspace-query-builder/factories/find-duplicates-query.factory.ts:129]: Do '%' work with equal statements? I didn't test locally yet but I would have guessed this doesn't work and we should just have eq: `${argsData[columnName]}` ?

**Comment by FelixMalfait** [comment]
Thank you!

**Review by magrinj — CHANGES_REQUESTED** [review]
Thanks for your PR, few minor comments :)

**Review by magrinj — APPROVED** [review]
LGTM ! Thanks for the changes and your contribution :)

**Pull request merged**